packages/main/src/plugin/troubleshooting.ts

```diff
@@ -0,0 +1,110 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import AdmZip from 'adm-zip';
+import moment from 'moment';
+import * as os from 'node:os';
+import * as fs from 'node:fs';
+
+export interface FileMap {
+  fileName: string;
+  content: string;
+}
+
+export type LogType = 'log' | 'warn' | 'trace' | 'debug' | 'error';
+
+export class Troubleshooting {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  constructor(private apiSender: any) {}
+
+  // Save a list of files with their filename and content to a zip file
+  async saveLogsToZip(fileMaps: FileMap[], destination: string): Promise<void> {
+    const zip = new AdmZip();
+
+    // Add files to the zip
+    fileMaps.forEach(fileMap => {
+      zip.addFile(`${fileMap.fileName}.txt`, Buffer.from(fileMap.content, 'utf8'));
+    });
+
+    // Write the zip file to the destination
+
+    // Only write if fileMaps is not empty
+    if (fileMaps.length > 0) {
+      zip.writeZip(destination);
+      console.log(`Zip file saved to ${destination}`);
+    }
+  }
+
+  getConsoleLogs(consoleLogs: { logType: LogType; message: string }[]): FileMap[] {
+    const content = consoleLogs.map(log => `${log.logType} : ${log.message}`).join('\n');
+    return [
+      {
+        fileName: generateLogFileName('console'),
+        content: content,
+      },
+    ];
+  }
+
+  // TODO: Implement any other logs (Windows & Linux) we wish to retrieve in the future.
+  async getSystemLogs(): Promise<FileMap[]> {
+    let logs: FileMap[] = [];
+    if (os.platform() === 'darwin') {
+      logs = await this.getMacSystemLogs();
+    } else if (os.platform() === 'win32') {
+      logs = await this.getWindowsSystemLogs();
+    } else {
+      console.log('Unsupported platform for retrieving system logs');
```

Comment:
I think we could "abstract" a log as either a file content or the output of a command. This way, we could have generic getFilesLogs() and getCommandLogs() functions, and in this if/elseif (or switch/case), we could just pass a list of files / commands.